utils/getServerSideProps.ts

```diff
@@ -1,22 +1,53 @@
 import { GetServerSidePropsContext, GetServerSidePropsResult } from 'next';
 import { validateRequest } from '~/backend/auth';
 import { User } from '~/shared/types';
+import { getUserSchema } from '~/shared/zodSchemas';
+
+const EMPTY_USER_OBJECT: User = {
+  email: '',
+  firstName: '',
+  lastName: '',
+  createdAt: new Date(0),
+  updatedAt: new Date(0),
+  uuid: '',
+  role: '',
+};
```

Comment:
Tein tyhjän objektin, jota käytetään kun käyttäjä ohjataan `login.tsx`-sivulle. Kun käyttäjä sinne ohjataan, ei tietokannasta ole löytynyt käyttäjän tietoja, ja `getServerSideProps`-funktio palauttaa pakosti `User`-typen. Siksi tein näin


Toinen tapa on pitää `login.tsx`:ssä oma `getServerSideProps`-funktio

Reply:
Parempi pitää `login.tsx`:ssä oma `getServerSideProps`-funktio, ks. kommentti siellä.

Reply:
Parempi pitää `login.tsx`:ssä oma `getServerSideProps`-funktio, ks. kommentti siellä.